ReactAndroid/src/main/java/com/facebook/react/views/image/ReactImageManager.java

```diff
@@ -93,9 +96,27 @@ public void setBorderWidth(ReactImageView view, float borderWidth) {
     view.setBorderWidth(borderWidth);
   }
 
-  @ReactProp(name = "borderRadius")
-  public void setBorderRadius(ReactImageView view, float borderRadius) {
-    view.setBorderRadius(borderRadius);
+  @ReactPropGroup(names = {
+      ViewProps.BORDER_RADIUS,
+      ViewProps.BORDER_TOP_LEFT_RADIUS,
+      ViewProps.BORDER_TOP_RIGHT_RADIUS,
+      ViewProps.BORDER_BOTTOM_RIGHT_RADIUS,
+      ViewProps.BORDER_BOTTOM_LEFT_RADIUS
+  }, defaultFloat = CSSConstants.UNDEFINED)
+  public void setBorderRadius(ReactImageView view, int index, float radius) {
+    if (!CSSConstants.isUndefined(radius)) {
+      radius = PixelUtil.toPixelFromDIP(radius);
+    }
+
+    if (index == 0) {
+      view.setBorderRadius(0, radius);
+      view.setBorderRadius(1, radius);
+      view.setBorderRadius(2, radius);
+      view.setBorderRadius(3, radius);
+      return;
+    }
+
+    view.setBorderRadius(index - 1, radius);
```

Comment:
So apparently this isn't correct:

React send prop updates as diffs (only sending props that have changed). Imagine you have component that has `borderRadius: 10, borderTopLeftRadius: 5`. You expect it to have border radius 10 everywhere except from top-left corder, where radius should be 5. Then when you update your component to have the following props: `borderRadius: 20, borderTopLeftRadius: 5`, react will only send update for "borderRadius" prop, which will reset also "topLeft" and will result in the component having radius set to 20 for each corner while you expect it to still have radius 5 for top-left corner